piptools/scripts/compile.py

```diff
@@ -381,13 +440,32 @@ def cli(
     config: Path | None,
     no_config: bool,
     constraint: tuple[str, ...],
+    build_deps_for_distributions: tuple[Literal["sdist", "wheel", "editable"], ...],
```

Comment:
Nitpick: this type could be put into a variable and reused. Don't consider this as a blocker, fixing it is optional.

Reply:
@apljungquist I think the combination of two is fine but I won't insist.

Reply:
I went with Option 1 because I think the fact that it is a tuple is incidental duplication; in both cases we would actually prefer to represent it as a set but it happens to be convenient that it is a tuple.

Reply:
Okay, marking this as resolved.